### DISCUSSION TITLE
Python enable function input to pass into query

### DISCUSSION THREAD

**Discussion by subaiwen** [duckdb/duckdb#6784] [Ideas]
Suppose I have dataframe:
```python
df_global = pd.DataFrame('x': [1,2,3]})
```
And I define a function:
```python
def foo(df):
    return duckdb.sql('select * from df').df()

foo(df_global)
```

Error: `CatalogException: Catalog Error: Table with name df does not exist!`

Is it possible to parse local objects in duckdb.sql()?

**Comment by Mytherin** [comment]
DuckDB does also look in the local scope - your example works fine for me:

```py
import pandas as pd
import duckdb
df_global = pd.DataFrame({'x': [1,2,3]})
def foo(df):
    return duckdb.sql('select * from df').df()

print(foo(df_global))
```

```bash
$ > python3 test.py
   x
0  1
1  2
2  3
```

Are you perhaps using an older version of DuckDB?

**Reply by subaiwen** [reply]
Sorry, i restate the issue i met here #6785

**Comment by Alex-Monahan** [comment]
You can also manually register a dataframe, which may help in this case!
https://duckdb.org/docs/api/python/data_ingestion#dataframes--arrow-tables
